tests/model/test_model.py

```diff
@@ -671,10 +678,31 @@ def test_append_message(self, mocker, user_profile, response,
         set_count.assert_called_once_with([response['id']], self.controller, 1)
 
         model.found_newest = False
+        model.notify_user.assert_called_once_with(response)
         model.append_message(event)
         # LOG REMAINS THE SAME IF UPDATE IS FALSE
         assert model.msg_list.log == log
 
+    @pytest.mark.parametrize('vary_each_msg, types_when_notify_called', [
+        ({'sender_id': 1}, []),  # model.user_id is 1
+        ({'sender_id': 2, 'flags': ['mentioned']}, ['stream', 'private']),
+        ({'sender_id': 2, 'flags': ['wildcard_mentioned']},
+         ['stream', 'private']),
+        ({'sender_id': 2, 'flags': []}, ['private']),
+    ], ids=['self_message', 'mentioned_msg', 'wildcard_mentioned',
+            'not_mentioned'])
+    def test_notify_users_calling_msg_type(self, mocker, model,
+                                           message_fixture, vary_each_msg,
+                                           types_when_notify_called):
+        message_fixture.update(vary_each_msg)
+        model.user_id = 1
+        notify = mocker.patch('zulipterminal.model.notify')
+        model.notify_user(message_fixture)
+        if message_fixture['type'] in types_when_notify_called:
+            assert notify.called
```

Comment:
I know we're limiting this test for now, so unless you want to include how notify is called, perhaps add a TODO here?